bakta/features/cds.py

```diff
@@ -770,93 +778,151 @@ def detect_pseudogenization_observations(alignment: str, ref_alignment: str, qst
+                observations[bc.PSEUDOGENE_EFFECT_START].add(pos)
+                observations['directions'].add(bc.FEATURE_END_3_PRIME)
+                log.info(
+                    'pseudogene observation: contig=%s, start=%i, stop=%i, strand=%s, original start=%i',
+                    cds['contig'], cds['start'], cds['stop'], cds['strand'], cds['start'] + pos
+                )
+            else:  # RBS was predicted (protein iso-form) -> skip
+                pass
+
+        # TODO implement case: point mutation -> loss of original start codon
+        if (alignment[0] == 'M' and ref_alignment[0] == 'M' and up_length != 0) and \
+                (alignment[up_length + 1] == 'M' and ref_alignment[up_length + 1] == 'M') and cds['rbs_motif'] is None:
+            # TODO correct structural annotation
+            pass
+
+    # Check alignment
     for char, ref_char in zip(alignment, ref_alignment):
-        if char == '\\':  # insertion
-            observations[bc.PSEUDOGENE_CAUSE_INSERTION].add(position)
-            observations['directions'].add(direction)
-            position += 1
+        if char == '-':
+            continue
+        elif char == '\\':  # insertion
+            insertions += 1
+            pos = get_abs_position(cds, start, position, edge)
+            observations[bc.PSEUDOGENE_CAUSE_INSERTION].add(pos)
+            observations['directions'].add(get_direction(position, edge))
             log.info(
-                'pseudogene observation: contig=%s, start=%i, stop=%i, strand=%s, cause=insertion, postion=%i',
-                cds['contig'], cds['start'], cds['stop'], cds['strand'], cds['start'] + position
+                'pseudogene observation: contig=%s, start=%i, stop=%i, strand=%s, cause=insertion, position=%i',
+                cds['contig'], cds['start'], cds['stop'], cds['strand'], pos
             )
+            position += 1
         elif char == '/':  # deletion
-            observations[bc.PSEUDOGENE_CAUSE_DELETION].add(position)
-            observations['directions'].add(direction)
+            deletions += 1
+            pos = get_abs_position(cds, start, position, edge)
+            observations[bc.PSEUDOGENE_CAUSE_DELETION].add(pos)
+            observations['directions'].add(get_direction(position, edge))
             log.info(
                 'pseudogene observation: contig=%s, start=%i, stop=%i, strand=%s, cause=deletion, position=%i',
-                cds['contig'], cds['start'], cds['stop'], cds['strand'], cds['start'] + position
+                cds['contig'], cds['start'], cds['stop'], cds['strand'], pos
             )
         elif char == '*':  # stop codon, selenocysteine, pyrolysine
             if ref_char == 'U':  # selenocysteine
-                observations[bc.PSEUDOGENE_EXCEPTION_SELENOCYSTEINE].add(position)
+                pos = get_abs_position(cds, start, position, edge)
+                observations[bc.PSEUDOGENE_EXCEPTION_SELENOCYSTEINE].add(pos)
```

Comment:
Give `position` and `pos` more descriptive and clear names:
1. rename `position` -> `alignment_position`
2. rename `pos` -> `genome_position`